src/cloudant/_common_util.py

```diff
@@ -160,7 +160,7 @@ def feed_arg_types(feed_type):
         return _COUCH_DB_UPDATES_ARG_TYPES
     return _CHANGES_ARG_TYPES
 
-def python_to_couch(options):
+def python_to_couch(options, encoder=None):
```

Comment:
AFAICT all `get_docs` calls have an encoder supplied which should in turn get here via https://github.com/cloudant/python-cloudant/blob/105fe2055562adbeb7c5ae41f97712794bf8716b/src/cloudant/_common_util.py#L252

However, I noticed that the view call to `get_docs` was slightly different in using the design document's encoder rather than the client one. 
https://github.com/cloudant/python-cloudant/blob/105fe2055562adbeb7c5ae41f97712794bf8716b/src/cloudant/view.py#L238

I think this is correct because if the design document has an encoder set it will be used in preference to the client one, but otherwise fall back to the client encoder if unset (via https://github.com/cloudant/python-cloudant/blob/105fe2055562adbeb7c5ae41f97712794bf8716b/src/cloudant/document.py#L68).

I think the caveat here is that someone who set an encoder specifically on a document (and no client level encoder) would also want to set that encoder on their design document before making view requests to guarantee the same encoder was used for the fields in the documents and the keys in the view query.

I don't think there is a problem compared to current behaviour but I wanted to call attention to it in case anyone had other thoughts.